src/cmsg.c

```diff
@@ -120,7 +115,7 @@ struct file_t recvfd(int sockfd)
 	/* TODO: Make this dynamic with MSG_PEEK. */
 	file.name = malloc(TAG_BUFFER);
 	if (!file.name)
-		error("recvfd: failed to allocate file.tag buffer");
+		errorf("recvfd: failed to allocate file.tag buffer: %m");
```

Comment:
`malloc()` failure doesn't set errno according to ISO C standard. The errno is set only on POSIX compliant systems. At this point I'm not sure if somebody would run conmon on non-POSIX systems but it's good to keep things portable if possisble I think. Having said that I'd just skip %m as "failed to allocate file.tag buffer" says it all if file.name is NULL.

Reply:
Thanks! I was on the verge whether to use %m or not, and there was a version without it :) went back to it